core/src/main/java/org/elasticsearch/search/sort/SortParseElement.java

```diff
@@ -156,6 +162,8 @@ private void addCompoundSortField(XContentParser parser, SearchContext context,
                                     missing = parser.textOrNull();
                                 } else if (context.parseFieldMatcher().match(innerJsonName, IGNORE_UNMAPPED)) {
                                     // backward compatibility: ignore_unmapped has been replaced with unmapped_type
+                                    deprecationLogger.deprecated("[" + IGNORE_UNMAPPED.getPreferredName() + "]  parameter will be removed"
```

Comment:
Since ParseField already logs when using a deprecated parameter, an easier way to do it would be to create the ParseField with something like `new ParseField("ignore_unmapped").withAllDeprecated("unmapped_type")`